rust/kernel/str.rs

```diff
@@ -167,6 +167,50 @@ impl CStr {
     pub const fn as_bytes_with_nul(&self) -> &[u8] {
         &self.0
     }
+
+    /// Yields a [`&str`] slice if the `CStr` contains valid UTF-8.
+    ///
+    /// If the contents of the `CStr` are valid UTF-8 data, this
+    /// function will return the corresponding [`&str`] slice. Otherwise,
+    /// it will return an error with details of where UTF-8 validation failed.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use kernel::str::CStr;
+    /// let cstr = CStr::from_bytes_with_nul(b"foo\0").unwrap();
+    /// assert_eq!(cstr.to_str(), Ok("foo"));
+    /// ```
+    #[inline]
+    pub fn to_str(&self) -> Result<&str, core::str::Utf8Error> {
+        core::str::from_utf8(self.as_bytes())
+    }
+
+    /// Unsafely convert this CStr into a `&str`, without checking for
+    /// valid UTF-8.
+    ///
+    /// # Safety
+    ///
+    /// Callers *must* ensure that CStr is valid UTF-8 before
+    /// calling this method. Converting a CStr into a `&str` that is
+    /// not valid UTF-8 is considered undefined behavior.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// # use kernel::c_str;
+    /// # use kernel::str::CStr;
+    /// // SAFETY: This is safe because string literals are guaranteed to be
+    /// // valid UTF-8 by the Rust compiler.
```

Comment:
Perhaps simplify here too, i.e. `SAFETY` comments are always about justifying why it is safe:

```suggestion
    /// // SAFETY: String literals are guaranteed to be valid UTF-8
    /// // by the Rust compiler.
```